CONTRIBUTING.md

```diff
@@ -1,9 +1,40 @@
 # Contributing to Atomic Reactor
 
 If you're reading this, it means that you want to contribute to Atomic Reactor. Great! Here are some tips to make sure your pull request gets accepted.
-Note: there are quite a few tools mentioned through this tutorial. To get them all, including Atomic Reactor dependencies, you can run `pip install -r requirements-devel.txt`.
+Note: there are quite a few tools mentioned through this tutorial. To get them all, including Atomic Reactor dependencies, see the [Environment preparation](#environment-preparation) section.
+
 Also note: while this seems to be an awfully long text, chances are you adhere to most of it just because you're a great developer. And don't worry! Us, core devs, must send code through pull requests, too (well, assuming it isn't an obvious oneliner).
 
+## Environment preparation
+
+While Atomic Reactor depends on the `docker-py` python package, one of its dependencies (docker-squash) requires the `docker` package to be installed.  Although docker and docker-py have different names in pypi, they are just different versions of the same package, i.e., they are installed under the same namespace. Hence, when we install one of these packages after the other, pip will overwrite files from the first installed package with files from the second one.
+
+For now, Atomic Reactor does not support the `docker` python package, so we want to keep the `docker-py` installation in our environment. For that, you must install docker-squash (which pulls docker) before docker-py, so the former will be overwritten by the latter.
+
+Note that just changing the order of the lists in the requirements file [does not guarantee the packages will be installed in that order](https://pip.pypa.io/en/stable/reference/pip_install/#installation-order).
+
+To make sure `docker` will get pulled before docker-py (and then overwritten by it), we can install our dependencies with
+
+```
+python setup.py develop
```

Comment:
When I run this on a fresh virtual env with python2, I get the following:
```
Removing /home/lucarval/.virtualenvs/spam/lib/python2.7/site-packages/docker_squash-1.0.7-py2.7.egg
Moving docker_squash-1.0.7-py2.7.egg to /home/lucarval/.virtualenvs/spam/lib/python2.7/site-packages
docker-squash 1.0.7 is already the active version in easy-install.pth
Installing docker-squash script to /home/lucarval/.virtualenvs/spam/bin

Installed /home/lucarval/.virtualenvs/spam/lib/python2.7/site-packages/docker_squash-1.0.7-py2.7.egg
error: The 'docker' distribution was not found and is required by docker-squash
```

To reproduce, be sure to create a virtual env with a unique name. Even if you remove the virtual env, some of it is cached if you recreate it with same name.

Reply:
If I rerun the command, it succeeds.
But I get about 8 test failures:
```python
>               assert schema in schemas
E               assert 1 in [2]

tests/plugins/test_koji_promote.py:896: AssertionError
```
Anyone else seeing this?

Reply:
Ah! So this one is solved by installing dockpulp.
Ok so I have a set of instructions that always work on first try in my env:

1. `pip install docker-squash`
2. `python setup.py develop`
3. `pip install -r requirements-devel.txt`
4. `pip install git+https://github.com/release-engineering/dockpulp`

3 and 4 can be combined, if we add `git+https://github.com/release-engineering/dockpulp` to `requirements-devel.txt`

Reply:
I could reproduce the `error: The 'docker' distribution was not found and is required by docker-squash` issue. I confirm that installing `python-squash` in **step 1** solves the problem.

The proposed steps work fine for python 2 environments.

I get 8 failing tests for python 3 when installing dockpulp:

```
>               assert schema in schemas
E               assert 1 in [2]

tests/plugins/test_koji_promote.py:896: AssertionError
```

Removing **step 4** for python 3 environments solves the issue. This means not installing dockpulp and therefore skipping some tests.

Reply:
That sounds fine. We don't run dockpulp plugins in python 3 in travis either. It may be that the plugins need some work to be python 3 compatible.

Reply:
Done